dll/cpl/input/input.c

```diff
@@ -78,7 +79,23 @@ SystemApplet(HWND hwnd, UINT uMsg, LPARAM lParam1, LPARAM lParam2)
     /* Advanced Settings */
     InitPropSheetPage(&page[1], IDD_PROPPAGEADVANCEDSETTINGS, AdvancedSettingsPageProc);
 
-    return (LONG)(PropertySheetW(&header) != -1);
+    ret = (LONG)(PropertySheetW(&header) != -1);
+
+    /* Reboot now? */
+    if (g_bRebootNeeded)
```

Comment:
Could you then adjust the following code: https://git.reactos.org/?p=reactos.git;a=blob;f=dll/cpl/input/settings_page.c;hb=dabb3f4d61da14bfb25c650a05f5943bea8ea4b4#l620
so that the reboot stuff stays centralized?

Reply:
I made it centralized.